src/qibo/models/error_mitigation.py

```diff
@@ -345,7 +345,16 @@ def closure():
 
         optimizer.step(closure)
         return params
-    return curve_fit(model, xdata, ydata, p0=params)[0]
+
+    if backend.platform in ["cupy", "cuquantum"]:  # pragma: no cover
+        # Currrently, ``cupy`` does not have compatibility with ``scipy.optimize``.
+        xdata = backend.to_numpy(xdata)
+        ydata = backend.to_numpy(ydata)
+        params = backend.to_numpy(params)
```

Comment:
I think that even for `tensorflow` a conversion to `numpy` is happening inside of `curve_fit` anyway. Therefore, we could even drop this check and just cast to `numpy` in all backends that are not `pytorch`. In any case, I am planning at some point to add a `curve_fit` method to the backends and get rid of this checks, thus you can safely ignore this.

Reply:
This is needed right now because tests are failing on GPU without this.